frame/staking/src/lib.rs

```diff
@@ -2143,12 +2146,13 @@ decl_module! {
 		/// # <weight>
 		/// See `crate::weight` module.
 		/// # </weight>
-		#[weight = T::WeightInfo::submit_solution_better(
+		#[weight = (
+			T::WeightInfo::submit_solution_better(
 			size.validators.into(),
 			size.nominators.into(),
 			compact.len() as u32,
 			winners.len() as u32,
-		)]
+		), frame_support::weights::DispatchClass::Operational)]
 		pub fn submit_election_solution_unsigned(
```

Comment:
the MaximumUnsignedWeight is not enforced here, shouldn't it be ?

Reply:
How can it even be enforced here? here we express how the weight is calculated. 

Then, using this formula, we craft a solution which is smaller than `MaximumUnsignedWeight `.

In other words, `MaximumUnsignedWeight ` is the target weight that the OCW will try and generate. An integrity check should make sure it is just less than the total block weight.

Reply:
So you're saying it's still fine to generate a heavier solution, it's just the default OCW won't generate one, correct? IMHO it's a bit counter-intuitive to do so, someone might assume that the solution is always smaller than `MaximumUnsignedWeight`, but that's not the case.

Reply:
> So you're saying it's still fine to generate a heavier solution, it's just the default OCW won't generate one, correct?

Sure, but I don't get what you suggest. How can I prevent someone from generating a heavier solution? they are free to do so, but it will be rejected here, based on this formula, if it is beyond the block limit. 

I think I am fundamentally missing your point here, please provide an example or code suggestion, will help I think :)

Reply:
I think it's more about future-proof naming than anything else. Since @thiolliere got impression that `MaximumUnsignedWeight` parameter is suppose to limit ANY solution that is received by the runtime he suggested to put a check here as well. 
The point of that PR however is to actually have OCW generate smaller solutions (there is not much point in restricting the solution here to be below any possible allowed weight), hence maybe this parameter should be renamed to reflect that.

Reply:
but this PR also makes the transaction able take operational weight, which wasn't the case before.
The way to limit would be to to compute weight again inside the extrinsic and return error if limit is overflowed. But anyway fine.